src/Microsoft.Azure.SignalR/ServerConnections/ConnectionMigrateOutFeature.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Microsoft.Azure.SignalR.ServerConnections
+{
+    class ConnectionMigrateOutFeature : IConnectionMigrateOutFeature

```

Comment:
What's this for?

Reply:
A connection can be either migrated-out or migrated-in, this feature will be added to a connection context when a connection is being migrated-out.

Reply:
So this is for OnDisconnectedAsync and the other is for OnConnectedAsync? Does it need to be 2 features?

Reply:
Yes, these 2 features for `connected` and `disconnected` respectively. It can be merged into 1 feature but I think it could be misled to our user?

Reply:
Since connection migrate-out could happen regardless of if a connection has been migrated-in.

Both a normal connection or a migrated connection could be migrated-out. Maybe our user would like to distinguish them in `OnDisconnectedAsync`

Reply:
Wouldn't that just mean 2 different properties?

Reply:
So we just leave the `MigrateFrom` to `null` when migration happened on this connection for the first time?